samples/openapi3/server/petstore/go/go-petstore/go/helpers.go

```diff
@@ -37,16 +37,16 @@ func IsZeroValue(val interface{}) bool {
 	return val == nil || reflect.DeepEqual(val, reflect.Zero(reflect.TypeOf(val)).Interface())
 }
 
-// AssertRecurseInterfaceRequired recursively checks each struct in a slice against the callback.
+// AssertRecurseInterface recursively checks each struct in a slice against the callback.
 // This method traverse nested slices in a preorder fashion.
-func AssertRecurseInterfaceRequired[T any](obj interface{}, callback func(T) error) error {
-	return AssertRecurseValueRequired(reflect.ValueOf(obj), callback)
+func AssertRecurseInterface[T any](obj interface{}, callback func(T) error) error {
```

Comment:
is it a must to rename this function? i'm afraid this may break the usage of our existing users

same for line 49

Reply:
Not must (the main reason I renamed it: now, both required and constraints use it)

Reply:
I suggest we keep the original/old name for the time being and rename it in the upcoming 8.0.x as some users do not like these kinda breaking changes in a minor release.

Reply:
another way is to keep the old function name (underlying it's calling the new function) and mark it as deprecated

Reply:
Done :)